### ISSUE TITLE
Any chance for Arm64 / aarch64?

### ISSUE THREAD

**Issue by mitchweaver** [aunetx/deezer-linux#37]
Title

With Asahi Linux becoming popular / usable would be nice to package for aarch64

**Comment by aunetx** [comment]
Hi, for flatpak I don't know how to build for arm, but if you want a hand-made solution you can build from source:
`make install_deps`
And then:
```sh
yarn --cwd run build-deb-arm
```
Which yields you an artifact in `artifacts`; you can change it to build snap, appimage, rpm, ... instead (see https://github.com/aunetx/deezer-linux/blob/master/package-append.json)

If this works for you, I will see about automatizing this an producing artifacts for both arm and amd64!

**Comment by aunetx** [comment]
@mitchweaver I added aarch64 support, and I plan to push it to flathub too BUT I can't test it myself (as I only have an amd64 laptop).

Could you test it from [the latest version](https://github.com/aunetx/deezer-linux/releases/tag/v5.30.580-3) (you can use the package you prefer, even appimage -- I just want to know if it works)?

**Comment by mitchweaver** [comment]
@aunetx  Sorry been a while missed this, looks like it does not

```
mitch@macbook[tmp] ./deezer-desktop-5.30.580-arm64.AppImage
Segmentation fault

mitch@macbook[tmp] cd deezer-desktop-5.30.580-arm64
mitch@macbook[deezer-desktop-5.30.580-arm64] ./deezer-desktop
Segmentation fault

mitch@macbook[deezer-desktop-5.30.580-arm64] uname -a
Linux macbook 6.4.11-401.asahi.fc38.aarch64+16k #1 SMP PREEMPT_DYNAMIC Mon Aug 21 19:38:10 UTC 2023 aarch64 GNU/Linux
```

**Comment by aunetx** [comment]
I'm sorry, I don't have any way to test it on arm... If someone wants to check what is wrong, I'm very open to PRs :)

**Comment by jnlmmln** [comment]
Hi everyone, is this maybe related to this: https://github.com/refi64/zypak/issues/25

> This is [fixed in recent Chromium versions](https://chromium-review.googlesource.com/c/chromium/src/+/3542265), included in Electron 19 and 18.3.7+. Any apps just need to upgrade their Electron versions to work, unfortunately.

What version of Electron is currently used? Is it `"electron": "^13.6.9",` which is set in `package-append.json`?

**Comment by aunetx** [comment]
@jnlmmln yes, it is electron 13 that is used. I am trying right now to upgrade to electron 28 (after all, it might work!), but if it is too high I will find a good version to work with.

**Comment by aunetx** [comment]
I can't get anything higher than electron 13 to work to be honest...

**Comment by eduarddejong** [comment]
I am no expert here, but is it possible to not package the electron13 within the same package, but configure it as a dependency instead?

I tried running it on my Raspberry Pi, which is also ARM64 based, and I got the same segmentation fault.
Obviously it's not working with an x86 binary.

**Comment by aunetx** [comment]
Hi, I think the electron versions have been upgraded now (should be 35+), and I heard that it was working (at least the version published on flathub), so closing this issue! Feel free to reopen it if the arm version still does not work (especially @eduarddejong as you were recently there :D)

**Comment by Deluxo** [comment]
Cross compiling from x64 to arm64 fails with this:

  @malept/flatpak-bundler 1> ========================================================================
  @malept/flatpak-bundler Building module deezer in /tmp/flatpak-bundler-99509-qYOOrTED8t1Z/.flatpak-builder/build/deezer-1
  @malept/flatpak-bundler ========================================================================
  @malept/flatpak-bundler  +8ms
  @malept/flatpak-bundler 1> Running: mkdir -p /app
  @malept/flatpak-bundler  +0ms
  @malept/flatpak-bundler 2> bwrap: execvp /bin/sh: No such file or directory
  @malept/flatpak-bundler  +812ms
  @malept/flatpak-bundler 2> Error: module deezer: Child process exited with code 1
  @malept/flatpak-bundler  +0ms
  ⨯ flatpak-builder failed with status code 1  failedTask=build stackTrace=Error: flatpak-builder failed with status code 1
    at ChildProcess.<anonymous> (/home/lukas/dev/deezer-linux/app/node_modules/@malept/flatpak-bundler/index.js:71:16)
    at ChildProcess.emit (node:events:519:28)
    at maybeClose (node:internal/child_process:1101:16)
    at Socket.<anonymous> (node:internal/child_process:456:11)
    at Socket.emit (node:events:519:28)
    at Pipe.<anonymous> (node:net:346:12)
make: *** [Makefile:106: build_flatpak_arm64] Error 1


tried googling, but to no avail.
Any ideas? Anybody?
